Implement listwatchresources API on the simulator (#165)
kubernetes-sigs/kube-scheduler-simulator: simulator/docs/api.md

## Watch the simulator's resources

Watch individual changes to all k8s resources in the simulator. This endpoint uses `Server-Sent Events`.
Once this API is called, the server will be continuously sending WatchEvent every time the event happens.

### HTTP Request

`GET /api/v1/listwatchresources`

#### Parameter
You can specify the `lastResourceVersion` of each resource, which can be retrieved using the `list` API of each resource.
If you won't specify it, this API calls the `list` and returns the result as "ADDED" Events before starting watch.  

We recommend to call the `list` API before the calling and to use `XXXLastResourceVersion` parameters.

The `ResourceVersion` must be treated as opaque by clients and passed unmodified back to the server.
See also [this page](https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions).

| parameter                | requirement | description                                                                                        |
|--------------------------|-------------|----------------------------------------------------------------------------------------------------|

> | podsLastResourceVersion  | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |

Reviewer: Other params as well.

```suggestion
| podsLastResourceVersion  | OPTIONAL    | If not specified, all resources are returned as `ADDED` Events first and then start to watch. |
```

| nodesLastResourceVersion | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |
| pvsLastResourceVersion   | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |
| pvcsLastResourceVersion  | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |
| scsLastResourceVersion   | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |
| pcsLastResourceVersion   | OPTIONAL    | If not specified, some existing resource data is returned as `ADDED` Events before starting watch. |

e.g.)
```

> /api/v1/watchresources?podsLastResourceVersion=213&nodesLastResourceVersion=213&pvsLastResourceVersion=213&pvcsLastResourceVersion=213&scsLastResourceVersion=213&pcsLastResourceVersion=213

Reviewer: ```suggestion
/api/v1/listwatchresources?podsLastResourceVersion=213&nodesLastResourceVersion=213&pvsLastResourceVersion=213&pvcsLastResourceVersion=213&scsLastResourceVersion=213&pcsLastResourceVersion=213
```
